SpringProject/src/test/java/com/gradproject2019/AuthUtilsTest.java

```diff
@@ -0,0 +1,59 @@
+package com.gradproject2019;
+
+import com.gradproject2019.users.service.UserServiceImpl;
+import com.gradproject2019.utils.AuthUtils;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class AuthUtilsTest {
+
+    @Test
+    public void shouldCheckPasswordIsValidAgainstRegex() {
```

Comment:
This is a testing antipattern - generally you shouldn't couple tests like this, if someone changed your regexs and one of these assertions were to fail, the whole test would fail and you'd have no idea which assertions were failing!

The solution to this would be to split out all of these tests into their own tests, with very descriptive names i.e. `shouldFailValidationIfPasswordHasNoSpecialCharacters" (applies to all fields)

Reply:
The point of AssertJ is that it gives you more detailed messages about which assertion failed, the test class will be verbose with all the tests split out. Don't think this is needed.

Reply:
Hi Zabi, I think we originally had our tests split out to test each particular regex requirement (i.e one for capital letters and another for special character but this resulted in a lot of duplicates. It probably because we weren't testing in the most effective way.
However I believe the above was constructed with guidance from Harry suggesting to remove individual tests and combine into one test. I understand what you're saying about testing each requirement in the regex, perhaps there is a better way to do this without duplication?